### PR TITLE
Feature/improve file manager

### PR DESCRIPTION
**Pull request by odeimaiz**
- The way file tree is created (recursiveness supported)
- Folder can be selected
- Has two outputs: file and folder

### CODE REVIEW

File: services/web/client/source/class/qxapp/components/workbench/widgets/FileManager.js [hunk 1]
```diff
@@ -85,36 +84,112 @@ qx.Class.define("qxapp.components.workbench.widgets.FileManager", {
...
-      root.add(tree1);
+      this.__getObjLists();
+    },
 
-      let tree2 = this.__userTree = this.__configureTreeItem(new qx.ui.tree.TreeFolder(), this.tr("User Files"));
-      tree2.setOpen(true);
-      root.add(tree2);
+    __getObjLists: function() {
+      /*
+      const objects = qxapp.data.Fake.getObjectList();
```
Comment (sanderegg) [inline services/web/client/source/class/qxapp/components/workbench/widgets/FileManager.js]: want to keep these comments?

File: services/web/client/source/class/qxapp/components/workbench/Workbench.js [hunk 2]
```diff
@@ -320,6 +320,13 @@ qx.Class.define("qxapp.components.workbench.Workbench", {
           let fileManager = new qxapp.components.workbench.widgets.FileManager();
           fileManager.addListener("FileSelected", function(data) {
             node.getMetadata().outputs[0].value = data.getData().filePath;
+            node.getMetadata().outputs[1].value = null;
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/components/workbench/Workbench.js]: addressing 'things' by array index seems a recipe for disaster

File: services/web/server/src/async_sio.py [hunk 3]
```diff
@@ -121,27 +121,21 @@ def stop_dynamic_service(sid, data):
...
 async def list_S3_objects(sid, data):
-    _LOGGER.debug("client %s requests S3 objects in %s", sid, data)
+    _LOGGER.debug("client %s requests objects in storage. Extra argument %s", sid, data)
     _config = s3_config()
 
     s3_client = S3Client(endpoint=_config.endpoint,
         access_key=_config.access_key, secret_key=_config.secret_key)
 
     objects = s3_client.list_objects_v2(_config.bucket_name)
+    dataOut = []
```
Comment (pcrespov) [inline services/web/server/src/async_sio.py]: python uses snake case ... not sure why the linter is not catching thisone ... I wonder if it is again deactivated!:(

File: services/web/server/src/async_sio.py [hunk 4]
```diff
@@ -121,27 +121,21 @@ def stop_dynamic_service(sid, data):
...
-    #objects = s3_client.list_objects_v2(data)
-    #for obj in objects:
-    #    dataOut = {}
-    #    dataOut['name'] = obj.object_name
-    #    dataOut['lastModified'] = json.dumps(obj.last_modified, indent=4, sort_keys=True, default=str)
-    #    dataOut['size'] = obj.size
-    #    await SIO.emit('listObjectsUser', data=dataOut, room=sid)
+        objInfo = {}
+        objInfo['path'] = obj.bucket_name + '/' + obj.object_name
+        objInfo['lastModified'] = json.dumps(obj.last_modified, indent=4, sort_keys=True, default=str)
```
Comment (pcrespov) [inline services/web/server/src/async_sio.py]: Is there a real need to format the json with indent and sorting keys? That is normally done to make it human-readable while here we intend to transmit it, right?

Comment (pcrespov) [reply services/web/server/src/async_sio.py]: @odeimaiz and what about this?

Comment (odeimaiz) [reply services/web/server/src/async_sio.py]: @pcrespov I will be happy to listen to other solutions

Comment (pcrespov) [reply services/web/server/src/async_sio.py]: @odeimaiz check for instance this https://gist.github.com/drmalex07/5149635e6ab807c8b21e

Comment (odeimaiz) [reply services/web/server/src/async_sio.py]: @pcrespov no worries, fixed

**Pull request merged**